sda/cmd/verify/verify.go

```diff
@@ -0,0 +1,303 @@
+					log.Errorf("failed so publish message, reason: %v", err.Error())
+				}
+
+				continue
+			}
+
+			var file database.FileInfo
+
+			file.Size, err = archive.GetFileSize(message.ArchivePath)
+
+			if err != nil {
+				log.Errorf("Failed to get archived file size, reson: %v", err.Error())
+
+				continue
+			}
+
+			archiveFileHash := sha256.New()
+			f, err := archive.NewFileReader(message.ArchivePath)
+			if err != nil {
+				log.Errorf("Failed to open archived file, reson: %v ", err.Error())
+
+				// Send the message to an error queue so it can be analyzed.
+				infoErrorMessage := broker.InfoError{
+					Error:           "Failed to open archived file",
+					Reason:          err.Error(),
+					OriginalMessage: message,
+				}
+
+				body, _ := json.Marshal(infoErrorMessage)
+				if err := mq.SendMessage(delivered.CorrelationId, conf.Broker.Exchange, "error", body); err != nil {
+					log.Errorf("failed to publish message, reason: (%v)", err.Error())
+				}
+
+				// Restart on new message
+				continue
+			}
+
+			hr := bytes.NewReader(header)
+			// Feed everything read from the archive file to archiveFileHash
+			mr := io.MultiReader(hr, io.TeeReader(f, archiveFileHash))
+
+			c4ghr, err := streaming.NewCrypt4GHReader(mr, *key, nil)
+			if err != nil {
+				log.Errorf("Failed to open c4gh decryptor stream, reson: %v", err.Error())
+
+				continue
+			}
+
+			md5hash := md5.New() // #nosec
+			sha256hash := sha256.New()
+
+			stream := io.TeeReader(c4ghr, md5hash)
+
+			if file.DecryptedSize, err = io.Copy(sha256hash, stream); err != nil {
+				log.Errorf("failed to copy decrypted data, reson: %v", err.Error())
+
+				// Send the message to an error queue so it can be analyzed.
+				infoErrorMessage := broker.InfoError{
+					Error:           "Failed to verify archived file",
+					Reason:          err.Error(),
+					OriginalMessage: message,
+				}
+
+				body, _ := json.Marshal(infoErrorMessage)
+				if e := mq.SendMessage(delivered.CorrelationId, conf.Broker.Exchange, "error", body); e != nil {
+					log.Errorf("Failed to publish error message: %v", e)
+				}
+
+				if err := delivered.Ack(false); err != nil {
+					log.Errorf("Failed to ack message: %v", err.Error())
+				}
+
+				continue
+			}
+
+			file.Checksum = archiveFileHash
+			file.DecryptedChecksum = sha256hash
+
+			//nolint:nestif
+			if !message.ReVerify {
+
+				c := schema.IngestionAccessionRequest{
+					User:     message.User,
+					FilePath: message.FilePath,
+					DecryptedChecksums: []schema.Checksums{
+						{Type: "sha256", Value: fmt.Sprintf("%x", sha256hash.Sum(nil))},
+						{Type: "md5", Value: fmt.Sprintf("%x", md5hash.Sum(nil))},
+					},
+				}
+
+				verifiedMessage, _ := json.Marshal(&c)
+
+				err = schema.ValidateJSON(fmt.Sprintf("%s/ingestion-accession-request.json", conf.Broker.SchemasPath), verifiedMessage)
+
+				if err != nil {
+					log.Errorf("Validation of outgoing (ingestion-accession-request) failed, reason: %v", err.Error())
```

Comment:
same question on log verbosity as above

Reply:
Any error here means that we have a broken container, since this is validation of the message *we* create.